packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -278,44 +282,48 @@ const DomainPicker: FunctionComponent< Props > = ( {
-									const isAvailable = availabilityStatus
-										? domainIsAvailableStatus?.includes( availabilityStatus )
-										: true;
-									return (
-										<SuggestionItem
-											key={ suggestion.domain_name }
-											isUnavailable={ ! isAvailable }
-											domain={ suggestion.domain_name }
-											cost={ suggestion.cost }
-											isLoading={
-												currentDomain === suggestion.domain_name && isCheckingDomainAvailability
-											}
-											hstsRequired={ suggestion.hsts_required }
-											isFree={ suggestion.is_free }
-											isRecommended={ isRecommended }
-											railcarId={ baseRailcarId ? `${ baseRailcarId }${ index }` : undefined }
-											onRender={ () =>
-												handleItemRender(
-													suggestion.domain_name,
-													`${ baseRailcarId }${ index }`,
-													index,
-													isRecommended
-												)
-											}
-											onSelect={ () => {
-												onDomainSelect( suggestion );
-											} }
-											selected={ currentDomain === suggestion.domain_name }
-											type={ itemType }
-										/>
-									);
-								} ) ?? times( quantity, ( i ) => <SuggestionItemPlaceholder key={ i } /> ) }
+								{ ( ! areDependenciesLoading &&
+									domainSuggestions?.map( ( suggestion, i ) => {
+										const index = existingSubdomain ? i + 1 : i;
+										const isRecommended = index === 1;
+										const availabilityStatus =
+											domainAvailabilities[ suggestion?.domain_name ]?.status;
+										// should availabilityStatus be falsy then we assume it is available as we have not checked yet.
+										const isAvailable = availabilityStatus
+											? domainIsAvailableStatus?.indexOf( availabilityStatus ) > -1
+											: true;
+										return (
+											<SuggestionItem
+												key={ suggestion.domain_name }
+												isUnavailable={ ! isAvailable }
+												domain={ suggestion.domain_name }
+												cost={ suggestion.cost }
+												isLoading={
+													currentDomain === suggestion.domain_name && isCheckingDomainAvailability
+												}
+												hstsRequired={ suggestion.hsts_required }
+												isFree={ suggestion.is_free }
+												isRecommended={ isRecommended }
+												railcarId={ baseRailcarId ? `${ baseRailcarId }${ index }` : undefined }
+												onRender={ () =>
+													handleItemRender(
+														suggestion.domain_name,
+														`${ baseRailcarId }${ index }`,
+														index,
+														isRecommended
+													)
+												}
+												onSelect={ () => {
+													onDomainSelect( suggestion );
+												} }
+												selected={ currentDomain === suggestion.domain_name }
+												type={ itemType }
+											/>
+										);
+									} ) ) ||
+									times( quantity - ( existingSubdomain ? 1 : 0 ), ( i ) => (
```

Comment:
Although I understand what this code is doing, it may not be very clear in the future / to a developer with little context.

We should either refactor the code to make it a bit more explicit or at least add a comment explaining that, while the dependencies are loading, we show a specific number of placeholder items.

Reply:
Great point.